packages/flutter/test/material/page_transitions_theme_test.dart

```diff
@@ -350,4 +350,56 @@ void main() {
     await tester.pumpAndSettle();
     expect(builtCount, 1);
   }, variant: TargetPlatformVariant.only(TargetPlatform.android));
+
+  testWidgets('android can use CupertinoPageTransitionsBuilder', (WidgetTester tester) async {
```

Comment:
can you also add a test for changing the theme during a gesture pop?

Reply:
This PR prevents breaking a gesture pop by keeping the `platform` unchanged during a gesture pop. If the `PageTransitionsTheme` is changed at this point, it should be invalid.🤔️

To avoid breaking a gesture pop, even if the `PageTransitionsTheme` is changed, the `PageTransitionsBuilder` should be kept unchanged, just like #99919.

Reply:
sorry I meant changing the default target platform, or Theme.platform

Reply:
I referenced the [back gesture while OS changes](https://github.com/flutter/flutter/blob/7d950864a1a01c1edd6270670cd62f0f0466132b/packages/flutter/test/material/page_test.dart#L497) and added a test named `back gesture while TargetPlatform changes`.